rtengine/ipwavelet.cc

```diff
@@ -1360,131 +1358,112 @@ void ImProcFunctions::ip_wavelet(LabImage * lab, LabImage * dst, int kall, const
+                                                const float logmax = log(tempmax); //log Max
                                                 //cp.sigmm change the "wider" of sigma
-                                                float rapX = (tempmean + siglh[level] * tempsig) / (tempmax); //rapport between sD / max
-                                                float inx = log(insigma);
-                                                float iny = log(rapX);
-                                                float rap = inx / iny; //koef
-                                                float asig = 0.166f / (tempsig * siglh[level]);
-                                                float bsig = 0.5f - asig * tempmean;
-                                                float amean = 0.5f / (tempmean);
+                                                const float rapX = (tempmean + siglh[level] * tempsig) / tempmax; //rapport between sD / max
+                                                const float inx = log(insigma);
+                                                const float iny = log(rapX);
+                                                const float rap = inx / iny; //koef
+                                                const float asig = 0.166f / (tempsig * siglh[level]);
+                                                const float bsig = 0.5f - asig * tempmean;
+                                                const float amean = 1.f / tempmean;
+
+                                                //equalizer for levels 0 1 and 3...  1.33 and 0.75 arbitrary values
+                                                float kcFactor = 1.f;
+                                                if(cp.denmet == 1) {
+                                                    if(level == 0 || level == 3) {
+                                                        kcFactor = 1.7f;
+                                                    }
+                                                } else if(cp.denmet == 2) {
+                                                    if(level == 0 || level == 3) {
+                                                        kcFactor = 0.3f;
+                                                    }
+                                                } else if(cp.denmet == 3) {
+                                                    if(level == 0 || level == 1) {
+                                                        kcFactor = 1.7f;
+                                                    }
+                                                } else if(cp.denmet == 4) {
+                                                    if(level == 0 || level == 1) {
+                                                        kcFactor = 0.3f;
+                                                    }
+                                                }
+                                                const float k = 1.f / siglh[level] > 1.f ? SQR(siglh[level]) : siglh[level];
```

Comment:
Any reason for not checking `sighl[level] < 1.f` here?

Reply:
Wasn't that comparison the other way around in
```c++
float k = siglh[level];
if(siglh[level] > 1.f) {
    k = SQR(siglh[level]);
}
```

Reply:
Ah, maybe add parentheses like
```c++
const float k = 1.f / (siglh[level] > 1.f ? SQR(siglh[level]) : siglh[level]);
```
(I have no idea how operator precedence works in this case...)

Reply:
Out of curiosity, I tried this and it appears to work out as
```c++
const float k = (1.f / siglh[level] > 1.f) ? SQR(siglh[level]) : siglh[level];
```
without the parentheses. (Same precedence, right-to-left evaluation.)